tests/loader_debug_ext_tests.cpp

```diff
@@ -0,0 +1,1063 @@
+ * deal in the Materials without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Materials, and to permit persons to whom the Materials are
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice(s) and this permission notice shall be included in
+ * all copies or substantial portions of the Materials.
+ *
+ * THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ *
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE MATERIALS OR THE
+ * USE OR OTHER DEALINGS IN THE MATERIALS.
+ *
+ * Author: Charles Giessen <charles@lunarg.com>
+ * Author: Mark Young <marky@lunarg.com>
+ */
+
+#include "test_environment.h"
+
+//
+// VK_EXT_debug_report
+//
+
+struct ExpectedDebugReportMessage {
+    bool any_message = false;
+    std::string message;
+    VkDebugReportObjectTypeEXT object_type;
+    VkDebugReportFlagBitsEXT flag;
+    bool found = false;
+};
+static ExpectedDebugReportMessage expected_debug_report{};
+
+static VkBool32 VKAPI_CALL test_DebugReportCallback(VkDebugReportFlagsEXT flags, VkDebugReportObjectTypeEXT objectType,
+                                                    uint64_t object, size_t location, int32_t messageCode, const char* pLayerPrefix,
+                                                    const char* pMessage, void* pUserData) {
+    if (objectType == expected_debug_report.object_type && 0 < (flags | expected_debug_report.flag)) {
+        std::string message_str = pMessage;
+        if (expected_debug_report.any_message || (std::string::npos != message_str.find(expected_debug_report.message))) {
+            expected_debug_report.found = true;
+        }
+    }
+    return VK_FALSE;
+}
+
+static void CreateDebugReportEnvironment(FrameworkEnvironment& env, VkDebugReportFlagsEXT flags,
+                                         VkInstanceCreateInfo& create_info) {
+    for (uint32_t icd = 0; icd < 3; ++icd) {
+        env.add_icd(TestICDDetails(TEST_ICD_PATH_VERSION_6, VK_API_VERSION_1_0));
+        env.get_test_icd(icd).physical_devices.push_back({});
+        env.get_test_icd(icd).physical_devices.push_back({});
+    }
+
+    // Initialize the expected output
+    expected_debug_report.any_message = false;
+    expected_debug_report.message = "";
+    expected_debug_report.object_type = VK_DEBUG_REPORT_OBJECT_TYPE_UNKNOWN_EXT;
+    expected_debug_report.found = false;
+
+    // Setup the debug report struct
+    static VkDebugReportCallbackCreateInfoEXT debug_report_info{VK_STRUCTURE_TYPE_DEBUG_REPORT_CALLBACK_CREATE_INFO_EXT};
```

Comment:
Again, leery of static but I see why its used (so its lifetime is longer than this function).

Just commenting to document my thoughts, not requesting changes.

Reply:
Done